src/t_zset.c

```diff
@@ -3312,26 +3404,22 @@ void genericZrangebylexCommand(client *c, int reverse) {
 
         /* If reversed, get the last node in range as starting point. */
         if (reverse) {
-            eptr = zzlLastInLexRange(zl,&range);
+            eptr = zzlLastInLexRange(zl,range);
         } else {
-            eptr = zzlFirstInLexRange(zl,&range);
+            eptr = zzlFirstInLexRange(zl,range);
         }
 
         /* No "first" element in the specified interval. */
         if (eptr == NULL) {
             addReply(c,shared.emptyarray);
-            zslFreeLexRange(&range);
```

Comment:
The ZRANGESTORE commands look like they emit an empty array instead of 0 when the zrange for some of the edge cases returns an empty array.

Reply:
yeah, indeed a bug. my next commit solves it and adds tests.